apps/oxlint/test/fixtures/sourceCode_scope_methods/output.snap.md

```diff
@@ -0,0 +1,154 @@
+ 1 | const topLevelConstant = 1,
+ 2 |   secondTopLevelConstant = 2;
+   :   ^^^^^^^^^^^^^^^^^^^^^^
+ 3 | 
+   `----
+
+  x scope-plugin(scope): getScope(topLevelFunction): type: function
+  | isStrict: true
+  | vars: [arguments, param, localConstant]
+  | through: [topLevelConstant, Math]
+  | upper: module
+  | 
+    ,-[files/index.js:4:1]
+  3 |     
+  4 | ,-> function topLevelFunction(param) {
+  5 | |     const localConstant = topLevelConstant + param;
+  6 | |     return function innerFunction() {
+  7 | |       return localConstant + Math.PI;
+  8 | |     }
+  9 | `-> }
+ 10 |     
+    `----
+
+  x scope-plugin(scope): isGlobalReference(topLevelFunction): false
+   ,-[files/index.js:4:10]
+ 3 | 
+ 4 | function topLevelFunction(param) {
+   :          ^^^^^^^^^^^^^^^^
+ 5 |   const localConstant = topLevelConstant + param;
+   `----
+
+  x scope-plugin(scope): isGlobalReference(param): false
+   ,-[files/index.js:4:27]
+ 3 | 
+ 4 | function topLevelFunction(param) {
+   :                           ^^^^^
+ 5 |   const localConstant = topLevelConstant + param;
+   `----
+
+  x scope-plugin(scope): getDeclaredVariables(): localConstant
+   ,-[files/index.js:5:3]
+ 4 | function topLevelFunction(param) {
+ 5 |   const localConstant = topLevelConstant + param;
+   :   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+ 6 |   return function innerFunction() {
+   `----
+
+  x scope-plugin(scope): isGlobalReference(localConstant): false
+   ,-[files/index.js:5:9]
+ 4 | function topLevelFunction(param) {
+ 5 |   const localConstant = topLevelConstant + param;
+   :         ^^^^^^^^^^^^^
+ 6 |   return function innerFunction() {
+   `----
+
+  x scope-plugin(scope): isGlobalReference(topLevelConstant): false
+   ,-[files/index.js:5:25]
+ 4 | function topLevelFunction(param) {
+ 5 |   const localConstant = topLevelConstant + param;
+   :                         ^^^^^^^^^^^^^^^^
+ 6 |   return function innerFunction() {
+   `----
+
+  x scope-plugin(scope): isGlobalReference(param): false
+   ,-[files/index.js:5:44]
+ 4 | function topLevelFunction(param) {
+ 5 |   const localConstant = topLevelConstant + param;
+   :                                            ^^^^^
+ 6 |   return function innerFunction() {
+   `----
+
+  x scope-plugin(scope): isGlobalReference(innerFunction): false
+   ,-[files/index.js:6:19]
+ 5 |   const localConstant = topLevelConstant + param;
+ 6 |   return function innerFunction() {
+   :                   ^^^^^^^^^^^^^
+ 7 |     return localConstant + Math.PI;
+   `----
+
+  x scope-plugin(scope): isGlobalReference(localConstant): false
+   ,-[files/index.js:7:12]
+ 6 |   return function innerFunction() {
+ 7 |     return localConstant + Math.PI;
+   :            ^^^^^^^^^^^^^
+ 8 |   }
+   `----
+
+  x scope-plugin(scope): isGlobalReference(Math): false
+   ,-[files/index.js:7:28]
+ 6 |   return function innerFunction() {
+ 7 |     return localConstant + Math.PI;
+   :                            ^^^^
+ 8 |   }
```

Comment:
I think this is a bug. The underlying cause is that scope manager isn't picking up references to global variables, leading to `isGlobalReference()` failing to search and find this reference. I'll have to investigate more tomorrow.

Reply:
Good news, it's not because of misalignment of AST.

It has to do with `@typescript-eslint`'s distinction between type variables and value variables. I noticed that references to `String` and `Object` are correctly identified as global but `Math` and `Function` aren't, and sure enough the latter are deemed type-only variables for some reason.

I still want to take some time to understand why exactly.